src/ks-clock-face.c

```diff
@@ -116,82 +118,113 @@ static void update_proc(Layer *layer, GContext *ctx) {
   };
 
   // Draw hands with positive length only
-  if(s_radius > 2 * HAND_MARGIN) {
+  if (s_radius > 2 * HAND_MARGIN) {
     graphics_draw_line(ctx, s_center, hour_hand);
   }
-  if(s_radius > HAND_MARGIN) {
+  if (s_radius > HAND_MARGIN) {
     graphics_draw_line(ctx, s_center, minute_hand);
   }
 }
 
-static void window_load(Window *window) {
-  Layer *window_layer = window_get_root_layer(window);
-  GRect window_bounds = layer_get_bounds(window_layer);
+static int prv_anim_percentage(AnimationProgress dist_normalized, int max) {
+  return (int)(float)(((float)dist_normalized / (float)ANIMATION_NORMALIZED_MAX) * (float)max);
+}
 
-  s_center = grect_center_point(&window_bounds);
+static void prv_radius_update(Animation *anim, AnimationProgress dist_normalized) {
+  s_radius = prv_anim_percentage(dist_normalized, FINAL_RADIUS);
 
-  s_canvas_layer = layer_create(window_bounds);
-  layer_set_update_proc(s_canvas_layer, update_proc);
-  layer_add_child(window_layer, s_canvas_layer);
+  layer_mark_dirty(s_canvas_layer);
 }
 
-static void window_unload(Window *window) {
-  layer_destroy(s_canvas_layer);
+static void prv_hands_update(Animation *anim, AnimationProgress dist_normalized) {
+  s_anim_time.hours = prv_anim_percentage(dist_normalized, prv_hours_to_minutes(s_last_time.hours));
+  s_anim_time.minutes = prv_anim_percentage(dist_normalized, s_last_time.minutes);
+
+  layer_mark_dirty(s_canvas_layer);
 }
 
-/*********************************** App **************************************/
+static void prv_start_animation() {
+  // Prepare animations
+  static AnimationImplementation s_radius_impl = {
+    .update = prv_radius_update
+  };
+  prv_animate(ANIMATION_DURATION, ANIMATION_DELAY, &s_radius_impl, false);
 
-static int anim_percentage(AnimationProgress dist_normalized, int max) {
-  return (int)(float)(((float)dist_normalized / (float)ANIMATION_NORMALIZED_MAX) * (float)max);
+  static AnimationImplementation s_hands_impl = {
+    .update = prv_hands_update
+  };
+  prv_animate(2 * ANIMATION_DURATION, ANIMATION_DELAY, &s_hands_impl, true);
 }
 
-static void radius_update(Animation *anim, AnimationProgress dist_normalized) {
-  s_radius = anim_percentage(dist_normalized, FINAL_RADIUS);
+static void prv_create_canvas() {
+  Layer *window_layer = window_get_root_layer(s_main_window);
+  GRect window_bounds = layer_get_unobstructed_bounds(window_layer);
```

Comment:
Do we want the unobstructed bounds, or the full window bounds here? 

(We should probably change the variable name if it is actually the unobstructed bounds)